staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -480,37 +489,43 @@ func (g *Cloud) getInstanceOrErrorsByNames(allNames []string) (map[string]*insta
 				continue
 			}
 			if found[inst.Name] != nil {
-				klog.Errorf("Instance name %q was duplicated (in zone %q and %q)", inst.Name, zone, found[inst.Name].Zone)
+				klog.Errorf("Instance name %q was duplicated (in zone %q and %q)\n", inst.Name, zone, found[inst.Name].instance.Zone)
 				continue
 			}
-			found[inst.Name] = &gceInstance{
-				Zone:  zone,
-				Name:  inst.Name,
-				ID:    inst.Id,
-				Disks: inst.Disks,
-				Type:  lastComponent(inst.MachineType),
+			if allNamesSet.Has(inst.Name) {
+				found[inst.Name] = &instanceOrError{
```

Comment:
the key in found (inst.Name) will be different from the names passed into this function?

Reply:
it should never be different - this is also outdated and this part has been removed.

Reply:
the reason I asked is I am thinking even if we want to return a map as previous commit, because of the code 
name = canonicalizeInstanceName(name)
the returned the map's keys seem different from the original names passed in?

Reply:
Oh sorry I misunderstood. Yes the key would be different. I cant find the rest of the code on github anymore but a complete solution would need to translate it back.

Reply:
have some concern about this name differences. When I checked your the original PR https://github.com/kubernetes/kubernetes/pull/79897/, it seems having the assumption these names must match.

Reply:
That needs to be fixed to reflect this. The assumption isnt here on this PR.